File: carpet_cleaner.client/src/app/main/order/model/OrderRecord.ts
import {MerchantRecord} from "../../../operator/model/MerchantRecord";
import {AddressRecord} from "../../../operator/model/AddressRecord";

export class OrderRecord {
  id:string;
  number:string;
  pickUpDate:Date;
  merchantList: MerchantRecord[];
  pickUpAddress: AddressRecord;

  public constructor(init?: Partial<OrderRecord>) {
    Object.assign(this, init);
  }
}

File: carpet_cleaner.controller/main/java/kz/nu/carpet_cleaner/controller/model/OrderRecord.java
package kz.nu.carpet_cleaner.controller.model;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class OrderRecord {
  public String id;
  public String customerId;
  public String number;
  public AddressRecord pickUpAddress;
  public List<MerchantRecord> merchantList = new ArrayList<>();
  public Date pickUpDate;
  public Date deliveryDate;
  public OrderStatus cleanStatus;
  public BigDecimal totalOrderPrice;
}
